DQMServices/Core/interface/MonitorElement.h

```diff
@@ -76,25 +96,42 @@ namespace dqm::impl {
     using Scalar = MonitorElementData::Scalar;
     using Kind = MonitorElementData::Kind;
 
+    // Comparison helper used in DQMStore to insert into sets. This needs deep
+    // private access to the MEData, that is why it lives here.
+    struct MEComparison {
+      using is_transparent = int;  // magic marker to allow C++14 heterogeneous set lookup.
+
+      auto make_tuple(MonitorElement *me) const {
+        return std::make_tuple(std::reference_wrapper(me->getPathname()), std::reference_wrapper(me->getName()));
+      }
+      auto make_tuple(MonitorElementData::Path const &path) const {
+        return std::make_tuple(path.getDirname(), path.getObjectname());
+      }
+      bool operator()(MonitorElement *left, MonitorElement *right) const {
+        return make_tuple(left) < make_tuple(right);
+      }
+      bool operator()(MonitorElement *left, MonitorElementData::Path const &right) const {
+        return make_tuple(left) < make_tuple(right);
+      }
+      bool operator()(MonitorElementData::Path const &left, MonitorElement *right) const {
+        return make_tuple(left) < make_tuple(right);
+      }
+      bool operator()(MonitorElementData::Path const &left, MonitorElementData::Path const &right) const {
+        return make_tuple(left) < make_tuple(right);
+      }
+    };
+
   protected:
     DQMNet::CoreObject data_;  //< Core object information.
-    // TODO: we only use the ::Value part so far.
-    // Still using the full thing to remain compatible with the new ME implementation.
 
-    std::atomic<MonitorElementData const *> frozen_;    // only set if this ME is in a product already
-    std::atomic<MutableMonitorElementData *> mutable_;  // only set if there is a mutable copy of this ME
+    std::atomic<MutableMonitorElementData *> mutable_;  // only set if this is a mutable copy of this ME
```

Comment:
Mostly because it is a left-over from the transparent-copy-on-write logic that lived here for a while (visible in red, was never actually used). As we settled on keeping the MEs in the `DQMStore` rather than products, I ripped it out to simplify things, but did not change the type.